app/scripts/lib/NameBridge.ts

```diff
@@ -0,0 +1,144 @@
+// The reason why we alias this is because we want to
+
+import {
+  NameController,
+  NameControllerState,
+  NameStateChange,
+  NameType,
+  SetNameRequest,
+} from '@metamask/name-controller';
+import { RestrictedControllerMessenger } from '@metamask/base-controller';
+import {
+  ApplyChangeCallback,
+  EntryComparators,
+  OneWayBridge,
+  TwoWayBridge,
+} from './EntryBridge';
+
+// Use the same type for both the source entries and the argument to NameController::setName.
+export type NameEntry = SetNameRequest;
+
+/**
+ * Returns a string key for the given name entry.
+ *
+ * @param entry
+ * @returns
+ */
+function getKey(entry: NameEntry): string {
+  return `${entry.type}/${entry.variation}/${entry.value}`;
+}
+
+const NameBridgeEntryComparators: EntryComparators<NameEntry> = {
+  getKey,
+  areEqual: (a, b) => getKey(a) === getKey(b) && a.name === b.name,
+};
+
+/**
+ * A one-way bridge between the name entries in another source and the name entries in
+ * the name controller.
+ *
+ * @param nameController - The name controller to update.
+ * @param getSourceEntries - A function that returns the name entries from the
+ * other source.
+ */
+export class OneWayNameBridge extends OneWayBridge<NameEntry> {
+  constructor(
+    nameController: NameController,
+    getSourceEntries: () => NameEntry[],
+  ) {
+    super({
+      comparators: NameBridgeEntryComparators,
+      getSourceEntries,
+      applyChangeToDestination: (type, entry) => {
+        if (type === 'deleted') {
+          nameController.setName({ ...entry, name: null });
+        } else {
+          nameController.setName(entry);
+        }
+      },
+    });
+  }
+}
+
+type AllowedEvents = NameStateChange;
+
+export type BridgeMessenger<Namespace extends string> =
```

Comment:
This is really nice technically, but for our purposes, we can just encapsulate and hardcode the `NameController` state subscription inside our `AbstractPetnamesBridge` given a specific `messenger` in the constructor argument.